.github/workflows/jewel_checks.yml

```
@@ -37,3 +39,54 @@ jobs:
       - name: Run ktfmt and ktlint checks on IDE plugin
         # We can only use static analysis that doesn't depend on compilation in the IDE plugin
         run: ./gradlew :samples:ide-plugin:ktfmtCheck :samples:ide-plugin:lintKotlinMain --continue --no-daemon
+
+  check_paths:
+    name: Check changed files
+    runs-on: ubuntu-latest
+    permissions:
+      pull-requests: read
+    outputs:
+      run_formalities: ${{ steps.filter.outputs.run_formalities }}
+
+    steps:
+      - uses: actions/checkout@v4
+        name: Check out repository
+
+      - uses: dorny/paths-filter@v3
+        name: Check which files are changed
+        id: filter
+        with:
+          filters: |
+            run_formalities:
+              - '!all(platform/jewel/ui/generated/org/jetbrains/jewel/ui/icons/AllIconsKeys.java)'
```

Should we add some condition to ensure we only run this when the user changes jewel files? Otherwise, it would run for all PRs, right?


```suggestion
      - uses: dorny/paths-filter@v3
        name: Check which files are changed
        id: filter
        with:
          predicate-quantifier: 'every'
          filters: |
            run_formalities:
              - 'platform/jewel/**'
              - '!all(platform/jewel/ui/generated/org/jetbrains/jewel/ui/icons/AllIconsKeys.java)'
```

---

Does it make sense to simplify to skip any generated folders inside jewel?

```
- '!all(platform/jewel/**/generated/**)'
```